### PR TITLE
Be st 0073

### PR DESCRIPTION
**Pull request by habibdevsy**
- create chat room when register user.
- fetch room id for specific user.

### CODE REVIEW

File: backend/src/Controller/ChatRoom/ChatRoomController.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+use OpenApi\Annotations as OA;
+use Nelmio\ApiDocBundle\Annotation\Security;
+
+/**
+ * fetch chat room.
+ * @Route("v1/chatroom/")
+ */
+class ChatRoomController extends BaseController
+{
+    private $chatRoomService;
```
Comment (ralsalloum) [inline backend/src/Controller/ChatRoom/ChatRoomController.php]: parameter type is required in parameter definition.

File: backend/src/Manager/ChatRoom/ChatRoomManager.php [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+use Doctrine\ORM\EntityManagerInterface;
+use DateTime;
+use App\Constant\ChatRoom\ChatRoomConstant;
+use Symfony\Component\Uid\Uuid;
+
+class ChatRoomManager
+{
+    private $autoMapping;
+    private $entityManager;
+    private $chatRoomRepository;
```
Comment (ralsalloum) [inline backend/src/Manager/ChatRoom/ChatRoomManager.php]: Class parameters' types need to be declared.

File: backend/src/Manager/ChatRoom/ChatRoomManager.php [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        $this->entityManager->flush();
+
+        return $chatRoom;
+    }
+
+    /**
+     * @param $userId
+     * @return ChatRoomEntity
+     */
+    public function createChatRoom($userId): ChatRoomEntity
```
Comment (ralsalloum) [inline backend/src/Manager/ChatRoom/ChatRoomManager.php]: Declare the type of the argument _userId_ in the function definition.

File: backend/src/Manager/ChatRoom/ChatRoomManager.php [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        $request->setUserId($userId);
+
+        return $this->create($request);
+    }
+
+    /**
+     * @param $userId
+     * @return ChatRoomEntity|null
+     */
+    public function getChatRoom($userId): ?ChatRoomEntity
```
Comment (ralsalloum) [inline backend/src/Manager/ChatRoom/ChatRoomManager.php]: Declare the type of the argument userId in the function definition.

File: backend/src/Manager/User/UserManager.php [hunk 5]
```diff
@@ -19,7 +20,7 @@ class UserManager
     private UserPasswordHasherInterface $encoder;
     private UserEntityRepository $userRepository;
 
-    public function __construct(AutoMapping $autoMapping, EntityManagerInterface $entityManager, UserPasswordHasherInterface $encoder, UserEntityRepository $userRepository)
+    public function __construct(AutoMapping $autoMapping, EntityManagerInterface $entityManager, UserPasswordHasherInterface $encoder, UserEntityRepository $userRepository, private ChatRoomManager $chatRoomManager)
```
Comment (ralsalloum) [inline backend/src/Manager/User/UserManager.php]: We need to define chatRoomManager as a class parameter, then initialize it in the constructor.

File: backend/src/Request/ChatRoom/ChatRoomCreateRequest.php [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+namespace App\Request\ChatRoom;
+
+class ChatRoomCreateRequest
+{
+    private int $userId;
+
+    /**
+     * Get the value of userId
+     */ 
+    public function getUserId()
```
Comment (ralsalloum) [inline backend/src/Request/ChatRoom/ChatRoomCreateRequest.php]: function return type need to be set.

File: backend/src/Request/ChatRoom/ChatRoomCreateRequest.php [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    {
+        return $this->userId;
+    }
+
+    /**
+     * Set the value of userId
+     *
+     * @return  self
+     */ 
+    public function setUserId($userId)
```
Comment (ralsalloum) [inline backend/src/Request/ChatRoom/ChatRoomCreateRequest.php]: Argument type need to be set.

File: backend/src/Service/ChatRoom/ChatRoomService.php [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+use App\AutoMapping;
+use App\Entity\ChatRoomEntity;
+use App\Manager\ChatRoom\ChatRoomManager;
+use App\Request\ChatRoom\ChatRoomCreateRequest;
+use App\Response\ChatRoom\ChatRoomResponse;
+
+class ChatRoomService
+{
+    private $autoMapping;
```
Comment (ralsalloum) [inline backend/src/Service/ChatRoom/ChatRoomService.php]: class parameters' types need to be declared.

**Review by ralsalloum — CHANGES_REQUESTED** [review]
Please resolve the requested changes.

File: backend/src/Controller/ChatRoom/ChatRoomController.php (hunk 1 above)
Comment (habibdevsy) [reply backend/src/Controller/ChatRoom/ChatRoomController.php]: It doesn't make sense to repeat the variable definition once in the comments and once within the builder, but I'll do it

**Comment by habibdevsy** [comment]
resolve the requested changes is done

**Review by ralsalloum — APPROVED** [review]

**Pull request merged**